mobile-app/app/screens/AppNavigator/screens/Loans/screens/PaybackLoanScreen.tsx

```diff
@@ -270,13 +273,13 @@ export function PaybackLoanScreen ({
         >
           <>
             <SetAmountButton
-              amount={new BigNumber(loanTokenAmount.amount ?? '0')}
+              amount={new BigNumber(loanTokenAmount.amount) > loanTokenBalanceInUSD ? loanTokenBalanceInUSD : new BigNumber('0')}
```

Comment:
Same here, use BigNumber.min ( BigNumber.max ( ), 0). Store it in a variable since they're same value